prow/jira/fake.go

```diff
@@ -0,0 +1,52 @@
+package jira
```

Comment:
Can we move this into its own subpackage prow/jira/fake so it doesn't get compiled into non-test binaries? Other than that lgtm